flowmachine/flowmachine/features/utilities/validators.py

```diff
@@ -0,0 +1,69 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from flowmachine.core import Query
+
+
+def valid_median_window(arg_name: str, value: int):
+    """
+    Raise a value error if the value provided is not an odd positive integer bigger than 1.
+    Parameters
+    ----------
+    arg_name : str
+        Argument name for error message
+    value : int
+        Value to check
+
+    Returns
+    -------
+    int
+        The validated value
+
+    Raises
+    ------
+    ValueError
+        If the value is even, negative, or less than 3
+
+    """
+    if value < 3:
+        raise ValueError(f"{arg_name} must be positive and greater than 1.")
+    if (value % 2) == 0:
```

Comment:
Perhaps best to switch the order of these validity checks (or change the first to `if value < 2`, or combine both into `if value < 3 or (value % 2) == 0`), otherwise `value=2` will give a confusing error message.